include/fiction/algorithms/physical_design/post_layout_optimization.hpp

```diff
@@ -0,0 +1,1238 @@
+                               add_fanout_to_route(op, fanouts_set.empty(), ffd);
+                               add_fanout_to_route(fanout, fanouts_set.empty(), ffd);
+
+                               // Continue until gate or primary output (PO) is found
+                               while (lyt.is_wire_tile(fanout) && lyt.fanout_size(lyt.get_node(fanout)) != 0 &&
+                                      lyt.fanout_size(lyt.get_node(fanout)) != 2)
+                               {
+                                   ffd.to_clear.push_back(fanout);
+                                   fanout = lyt.outgoing_data_flow(fanout).front();
+
+                                   // Add fanout to the respective route
+                                   add_fanout_to_route(fanout, fanouts_set.empty(), ffd);
+                               }
+
+                               // Set the respective fanout based on the route
+                               if (fanouts_set.empty())
+                               {
+                                   fanout1 = fanout;
+                               }
+                               else
+                               {
+                                   fanout2 = fanout;
+                               }
+
+                               fanouts_set.insert(fanout);
+                           }
+                       });
+
+    // add fanins and fanouts if existing
+    if (!fanin1.is_dead())
+    {
+        ffd.fanins.push_back(fanin1);
+    }
+    if (!fanin2.is_dead())
+    {
+        ffd.fanins.push_back(fanin2);
+    }
+    if (!fanout1.is_dead())
+    {
+        ffd.fanouts.push_back(fanout1);
+    }
+    if (!fanout2.is_dead())
+    {
+        ffd.fanouts.push_back(fanout2);
+    }
+
+    return ffd;
+}
+
+/**
+ * @brief Gets a path between two coordinates and obstructs the tiles along the path.
+ *
+ * This helper function computes a path between two coordinates using the A* algorithm.
+ * It then obstructs the tiles along the path in the given layout.
+ *
+ * @tparam Lyt Gate-level layout.
+ *
+ * @param lyt Reference to the layout.
+ * @param start The starting coordinate of the path.
+ * @param end The ending coordinate of the path.
+ * @param distance The distance functor used by the A* algorithm.
+ * @param cost_func The cost functor used by the A* algorithm.
+ * @param params The A* search parameters.
+ *
+ * @return The computed path as a sequence of coordinates in the layout.
+ */
+template <typename Lyt>
+layout_coordinate_path<Lyt> get_path_and_obstruct(Lyt& lyt, const coordinate<Lyt>& start, const coordinate<Lyt>& end,
+                                                  const twoddwave_distance_functor<Lyt, uint64_t>& distance,
+                                                  const unit_cost_functor<Lyt, uint8_t>&           cost_func,
+                                                  const a_star_params&                             params)
```

Comment:
Correct me if I'm wrong here, but don't you use the same distance and cost function and even the same parameters always? Couldn't you then pass on the parameters here?